internal/workflow.go

```diff
@@ -383,6 +383,15 @@ func NewFuture(ctx Context) (Future, Settable) {
 	return impl, impl
 }
 
+func (wc *workflowEnvironmentInterceptor) ProcessSignal(Context, string, interface{}) {
+	// no op
```

Comment:
You sure? If it's the same as the root interceptor in Java, I would expect the logic that actually triggers user code processing the signal be around here. Or scheduling of this processing by creating a thread (to respect the ordering).